worlds/subnautica/__init__.py

```diff
@@ -72,44 +87,62 @@ def generate_basic(self):
         # Victory item
         self.world.get_location("Aurora - Captain Data Terminal", self.player).place_locked_item(
             neptune_launch_platform)
-        self.world.get_location("Neptune Launch", self.player).place_locked_item(
-            SubnauticaItem("Victory", ItemClassification.progression, None, player=self.player))
-
-    def set_rules(self):
-        set_rules(self.world, self.player)
-
-    def create_regions(self):
-        create_regions(self.world, self.player)
+        for event in Locations.events:
+            self.world.get_location(event, self.player).place_locked_item(
+                SubnauticaItem(event, ItemClassification.progression, None, player=self.player))
+        # make the goal event the victory "item"
+        self.world.get_location(self.world.goal[self.player].get_event_name(), self.player).item.name = "Victory"
+
+    def fill_slot_data(self) -> Dict[str, Any]:
+        goal: Options.Goal = self.world.goal[self.player]
+        item_pool: Options.ItemPool = self.world.item_pool[self.player]
+        vanilla_tech: List[str] = []
+        if item_pool == Options.ItemPool.option_valuable:
+            for item in Items.item_table.values():
+                if item["classification"] == ItemClassification.filler:
+                    vanilla_tech.append(item["tech_type"])
+
+        slot_data: Dict[str, Any] = {
+            "goal": goal.current_key,
+            "vanilla_tech": vanilla_tech,
+        }
 
-    def fill_slot_data(self):
-        slot_data = {}
         return slot_data
 
     def create_item(self, name: str) -> Item:
-        item = lookup_name_to_item[name]
-        return SubnauticaItem(name,
-                              ItemClassification.progression if item["progression"] else ItemClassification.filler,
-                              item["id"], player=self.player)
-
-
-def create_region(world: MultiWorld, player: int, name: str, locations=None, exits=None):
-    ret = Region(name, RegionType.Generic, name, player)
-    ret.world = world
-    if locations:
-        for location in locations:
-            loc_id = locations_lookup_name_to_id.get(location, 0)
-            location = SubnauticaLocation(player, location, loc_id, ret)
-            ret.locations.append(location)
-    if exits:
-        for exit in exits:
-            ret.exits.append(Entrance(player, exit, ret))
+        item_id: int = self.item_name_to_id[name]
```

Comment:
wouldn't it be better to add type info to the dict?

Reply:
type info is in the dict as well.

Reply:
my point is that you don't have to add type hints to local variables if they get assigned from a fully typed source, but it doesn't hurt either